core/store/orm/orm.go

```diff
@@ -740,17 +740,18 @@ func (orm *ORM) FindEthTxAttempt(hash common.Hash) (*models.EthTxAttempt, error)
 	return ethTxAttempt, nil
 }
 
-// GetLastNonce retrieves the last known nonce in the database for an account
-func (orm *ORM) GetLastNonce(address common.Address) (uint64, error) {
+// GetNextNonce retrieves the next_nonce for a key
+func (orm *ORM) GetNextNonce(address common.Address) (uint64, error) {
 	orm.MustEnsureAdvisoryLock()
-	var transaction models.EthTx
-	err := orm.DB.Order("nonce desc").Where(`"from_address" = ?`, address).First(&transaction).Error
-	if err == gorm.ErrRecordNotFound {
-		return 0, nil
-	} else if err != nil {
+	var key models.Key
+	err := orm.DB.First(&key, "address = ?", address).Error
+	if err != nil {
```

Comment:
Do we ensure a record always exists somewhere else?

Reply:
Ah good call.